In [ ]:
import pandas as pd

In [ ]:
from google.colab import drive

drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
folder_path = '/content/drive/MyDrive/INFO 4390 Project/'

df = pd.read_csv(folder_path + 'raw_data.csv')

# Data Cleaning

The first step in our data cleaning is to rename the variables to more recognizeable words and removing all 'NaN' values. There is enough data in this dataset and few anough NaN values that this will not skew our data in any one direction by just removing these rows. We are also dropping the 'type' column because all of the universities in this dataset are 4-year.

In [ ]:
df = df.rename(columns = {"fips_ipeds" : "state", "inst_name" : "college", "slevel" : "type", "fourcat" : "selectivity"})
df[["col_white", "col_black", "col_hispa", "col_asian", "col_amind", "col_pacis", "col_twora", "dif_white", "dif_black", "dif_hispa", "dif_asian", "dif_amind", "dif_pacis", "dif_twora"]] = df[["col_white", "col_black", "col_hispa", "col_asian", "col_amind", "col_pacis", "col_twora", "dif_white", "dif_black", "dif_hispa", "dif_asian", "dif_amind", "dif_pacis", "dif_twora"]].fillna(0.0)
df.drop(columns=['type'])

unitid  year    state                   college            selectivity  \
0      100654  2009  Alabama  Alabama A & M University       Public Selective   
1      100654  2010  Alabama  Alabama A & M University       Public Selective   
2      100654  2011  Alabama  Alabama A & M University       Public Selective   
3      100654  2012  Alabama  Alabama A & M University       Public Selective   
4      100654  2013  Alabama  Alabama A & M University       Public Selective   
...       ...   ...      ...                       ...                    ...   
17262  240727  2013  Wyoming     University of Wyoming  Public More Selective   
17263  240727  2014  Wyoming     University of Wyoming  Public More Selective   
17264  240727  2015  Wyoming     University of Wyoming  Public More Selective   
17265  240727  2016  Wyoming     University of Wyoming  Public More Selective   
17266  240727  2017  Wyoming     University of Wyoming  Public More Selective   

       public  private  forprofit  selective  more_selective  ...  dif_asian  \
0           1        0          0          1               0  ...  -0.741711   
1           1        0          0          1               0  ...  -0.895135   
2           1        0          0          1               0  ...  -0.953688   
3           1        0          0          1               0  ...  -1.062909   
4           1        0          0          1               0  ...  -0.952295   
...       ...      ...        ...        ...             ...  ...        ...   
17262       1        0          0          0               1  ...   0.281356   
17263       1        0          0          0               1  ...   0.348824   
17264       1        0          0          0               1  ...   0.207345   
17265       1        0          0          0               1  ...   0.141128   
17266       1        0          0          0               1  ...   0.044605   

       col_amind  mkt_amind  dif_amind  col_pacis  mkt_pacis  dif_pacis  \
0       0.200758   0.564552  -0.363794   0.000000   0.155576   0.000000   
1       0.060852   0.581733  -0.520881   0.000000   0.089841  -0.089841   
2       0.072957   0.499614  -0.426656   0.024319   0.047023  -0.022703   
3       0.049008   0.502863  -0.453855   0.024504   0.035682  -0.011178   
4       0.123426   0.615549  -0.492122   0.098741   0.044647   0.054094   
...          ...        ...        ...        ...        ...        ...   
17262   0.662052   2.563285  -1.901233   0.290902   0.050326   0.240575   
17263   0.565086   2.498113  -1.933027   0.292634   0.214747   0.077887   
17264   0.609632   2.397501  -1.787868   0.243853   0.380927  -0.137074   
17265   0.613178   2.288063  -1.674884   0.207857   0.416011  -0.208154   
17266   0.498805   2.544634  -2.045829   0.166268   0.530276  -0.364007   

       col_twora  mkt_twora  dif_twora  
0       0.000000   1.528652   0.000000  
1       0.000000   1.654546  -1.654546  
2       0.000000   1.617237  -1.617237  
3       0.000000   1.604016  -1.604016  
4       0.000000   1.790837  -1.790837  
...          ...        ...        ...  
17262   2.206841   3.420510  -1.213668  
17263   2.603431   3.504740  -0.901309  
17264   3.098963   3.475123  -0.376159  
17265   3.440033   3.091336   0.348697  
17266   3.938481   3.367595   0.570886  

[17267 rows x 33 columns]

Now, for our models, we want the **average race percentage statistics** over the years provided for each university. We will create a new dataframe that has these average percentages for each of the race groups `col_*race_group*`. The new df is grouped by state and college.

In [ ]:
columns_to_include = ["state", "college", "public", "private", "selective", "more_selective", "non_selective", "total_enrollment"]
avg_columns = ["col_white", "col_black", "col_hispa", "col_asian", "col_amind", "col_pacis", "col_twora", "dif_white", "dif_black", "dif_hispa", "dif_asian", "dif_amind", "dif_pacis", "dif_twora", "mkt_white", "mkt_black", "mkt_hispa", "mkt_asian", "mkt_amind", "mkt_pacis", "mkt_twora", 'public', 'private', 'selective', 'more_selective', 'non_selective', 'total_enrollment']
columns_to_include.extend(avg_columns)
df2 = df.groupby(['state', 'college'])[avg_columns].mean().reset_index()
df2

state                             college  col_white  col_black  \
0       Alabama            Alabama A & M University   2.721590  93.950898   
1       Alabama            Alabama State University   1.319264  94.225080   
2       Alabama                  Amridge University  30.783844  33.924744   
3       Alabama             Athens State University  78.428936  11.972484   
4       Alabama                   Auburn University  84.445796   7.037638   
...         ...                                 ...        ...        ...   
2431  Wisconsin    University of Wisconsin-Superior  82.795638   1.657170   
2432  Wisconsin  University of Wisconsin-Whitewater  85.008887   4.605289   
2433  Wisconsin                  Viterbo University  90.415692   1.823594   
2434  Wisconsin          Wisconsin Lutheran College  83.953024   5.098209   
2435    Wyoming               University of Wyoming  78.310024   1.127528   

      col_hispa  col_asian  col_amind  col_pacis  col_twora  dif_white  ...  \
0      0.568847   0.183286   0.155261   0.097960   0.833909 -59.900314  ...   
1      0.883264   0.163126   0.079592   0.052987   0.610119 -54.552670  ...   
2      2.437949   0.718030   0.113526   0.062112   0.000000 -24.717332  ...   
3      1.743141   0.733095   1.950308   0.063228   1.268810  15.562983  ...   
4      2.893473   2.186667   0.650116   0.011044   0.457418  25.724310  ...   
...         ...        ...        ...        ...        ...        ...  ...   
2431   1.969304   1.141414   2.266822   0.092822   2.496747  -7.024135  ...   
2432   4.499936   1.639131   0.211966   0.057065   2.935055  13.240540  ...   
2433   2.274889   1.759734   0.535673   0.029761   1.210252   1.980485  ...   
2434   4.533477   1.077365   0.578613   0.064289   1.699697  33.975082  ...   
2435   5.946050   1.164442   0.705122   0.207030   2.503126  -0.139379  ...   

      mkt_asian  mkt_amind  mkt_pacis  mkt_twora  public  private  selective  \
0      1.113726   0.559253   0.052924   1.919729     1.0      0.0        1.0   
1      1.128831   0.337028   0.037316   1.409747     1.0      0.0        1.0   
2      1.118081   0.313141   0.060274   1.544599     0.0      1.0        0.0   
3      1.150877   0.509181   0.046996   1.790873     1.0      0.0        0.0   
4      1.274598   0.528624   0.043030   1.703866     1.0      0.0        0.0   
...         ...        ...        ...        ...     ...      ...        ...   
2431   2.674915   1.862803   0.121511   1.699608     1.0      0.0        1.0   
2432   3.686060   0.722951   0.044428   2.756883     1.0      0.0        1.0   
2433   2.478425   0.526834   0.083495   1.601961     0.0      1.0        1.0   
2434   4.541282   0.361789   0.035315   2.601499     0.0      1.0        1.0   
2435   1.000910   2.472015   0.202552   3.143894     1.0      0.0        0.0   

      more_selective  non_selective  total_enrollment  
0                0.0            0.0       4405.666667  
1                0.0            0.0       4795.555556  
2                0.0            1.0        355.800000  
3                0.0            1.0       3082.666667  
4                1.0            0.0      21013.250000  
...              ...            ...               ...  
2431             0.0            0.0       2405.000000  
2432             0.0            0.0      10301.777778  
2433             0.0            0.0       2004.555556  
2434             0.0            0.0        966.222222  
2435             1.0            0.0       9829.888889  

[2436 rows x 29 columns]

In [ ]:
df2.columns

Index(['state', 'college', 'col_white', 'col_black', 'col_hispa', 'col_asian',
       'col_amind', 'col_pacis', 'col_twora', 'dif_white', 'dif_black',
       'dif_hispa', 'dif_asian', 'dif_amind', 'dif_pacis', 'dif_twora',
       'mkt_white', 'mkt_black', 'mkt_hispa', 'mkt_asian', 'mkt_amind',
       'mkt_pacis', 'mkt_twora', 'public', 'private', 'selective',
       'more_selective', 'non_selective', 'total_enrollment'],
      dtype='object')


In examining the data, specifically the `col_*race_group*` columns, we noticed that the percentages do not always sum up to 100. While the dataset dictionary and the website do not specify why this is, it could be for a number of reasons. One possibility is that the percentages are rounded or truncated, leading to slight discrepancies when summing them up. Another possibility is that the dataset may include individuals who did not report their race or ethnicity, resulting in missing data that could affect the overall percentage calculations. Additionally, variations in data collection methods, reporting practices, and demographic changes over time could also contribute to these discrepancies.

Below is a summary of the different sums.

In [ ]:
cols_to_sum = [col for col in df2.columns if col.startswith('col_')]
df2['sum_races'] = df2[cols_to_sum].sum(axis=1)
df2['sum_races'].describe()

count    2436.000000
mean       89.391486
std        11.154996
min         0.000000
25%        87.052456
50%        92.836996
75%        96.317414
max       100.000008
Name: sum_races, dtype: float64

As we can see, the average of the sums is approximately 89.4%, which is about 10% less than 100%. Given these statistics on the sums, our group has decided to do the follow to ensure the percentages sum to 100%:

1. If the sum of the percentages is below 90%, it will be deleted.
2. If the sum of the percentages is above 90% then each of the `col_*race_group*` values for that row will be scaled up to 100%.

Doing this will ensure that the racial percentages for each row accurately represent the distribution of races within the corresponding institution while maintaining consistency with the total percentage sum of 100%. By adhering to these criteria, we aim to mitigate potential inaccuracies resulting from rounding errors or data inconsistencies. This approach not only maintains the integrity of the dataset but also ensures that the analysis and interpretations drawn from it are robust and reliable. Additionally, since our dataset has so much data anyways, there is still plenty to work with for our future analyses.

In [ ]:
# remove columns where sum_races is below 90%
filtered_df = df2[df2['sum_races'] >= 90]

# If sum_races is above 90%, scale the race columns for each row so that they sum to 100%
for col in ['col_white', 'col_black', 'col_hispa', 'col_asian', 'col_amind', 'col_pacis', 'col_twora']:
    filtered_df[col] = filtered_df[col] / filtered_df['sum_races'] * 100

filtered_df = filtered_df.drop(columns=['sum_races'])

filtered_df

<ipython-input-8-fdf8075a96d5>:6: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  filtered_df[col] = filtered_df[col] / filtered_df['sum_races'] * 100


state                             college  col_white  col_black  \
0       Alabama            Alabama A & M University   2.762706  95.370244   
1       Alabama            Alabama State University   1.355407  96.806491   
3       Alabama             Athens State University  81.560872  12.450587   
4       Alabama                   Auburn University  86.449566   7.204630   
5       Alabama       Auburn University Main Campus  87.406235   7.883860   
...         ...                                 ...        ...        ...   
2430  Wisconsin       University of Wisconsin-Stout  91.003790   1.327253   
2431  Wisconsin    University of Wisconsin-Superior  89.586359   1.793087   
2432  Wisconsin  University of Wisconsin-Whitewater  85.904589   4.653813   
2433  Wisconsin                  Viterbo University  92.214243   1.859869   
2434  Wisconsin          Wisconsin Lutheran College  86.545339   5.255633   

      col_hispa  col_asian  col_amind  col_pacis  col_twora  dif_white  ...  \
0      0.577441   0.186055   0.157606   0.099440   0.846508 -59.900314  ...   
1      0.907462   0.167595   0.081772   0.054439   0.626834 -54.552670  ...   
3      1.812751   0.762370   2.028190   0.065753   1.319478  15.562983  ...   
4      2.962130   2.238553   0.665542   0.011306   0.468272  25.724310  ...   
5      2.214625   1.847222   0.648058   0.000000   0.000000  25.340940  ...   
...         ...        ...        ...        ...        ...        ...  ...   
2430   2.156830   3.286284   0.378542   0.105612   1.741688  -1.497967  ...   
2431   2.130822   1.235030   2.452742   0.100435   2.701525  -7.024135  ...   
2432   4.547350   1.656402   0.214200   0.057667   2.965980  13.240540  ...   
2433   2.320141   1.794738   0.546329   0.030353   1.234327   1.980485  ...   
2434   4.673462   1.110632   0.596479   0.066274   1.752180  33.975082  ...   

      mkt_asian  mkt_amind  mkt_pacis  mkt_twora  public  private  selective  \
0      1.113726   0.559253   0.052924   1.919729     1.0      0.0        1.0   
1      1.128831   0.337028   0.037316   1.409747     1.0      0.0        1.0   
3      1.150877   0.509181   0.046996   1.790873     1.0      0.0        0.0   
4      1.274598   0.528624   0.043030   1.703866     1.0      0.0        0.0   
5      1.134392   0.460121   0.098045   1.245337     1.0      0.0        0.0   
...         ...        ...        ...        ...     ...      ...        ...   
2430   2.810809   1.067838   0.070899   1.509660     1.0      0.0        1.0   
2431   2.674915   1.862803   0.121511   1.699608     1.0      0.0        1.0   
2432   3.686060   0.722951   0.044428   2.756883     1.0      0.0        1.0   
2433   2.478425   0.526834   0.083495   1.601961     0.0      1.0        1.0   
2434   4.541282   0.361789   0.035315   2.601499     0.0      1.0        1.0   

      more_selective  non_selective  total_enrollment  
0                0.0            0.0       4405.666667  
1                0.0            0.0       4795.555556  
3                0.0            1.0       3082.666667  
4                1.0            0.0      21013.250000  
5                1.0            0.0      19918.000000  
...              ...            ...               ...  
2430             0.0            0.0       8102.666667  
2431             0.0            0.0       2405.000000  
2432             0.0            0.0      10301.777778  
2433             0.0            0.0       2004.555556  
2434             0.0            0.0        966.222222  

[1596 rows x 29 columns]

As we can see, there are still 1605 data points, which is plenty for our models.

For our models in phase3_code, we need to create one more new columns. We need to create `col_poc`, which represents the percentage of students who are POC, or Persons of Color. This is just created by summing up `col_black`,	`col_hispa`,	`col_asian`,	`col_amind`,	`col_pacis`, and `col_twora`.

In [ ]:
filtered_df['col_poc'] = filtered_df[['col_black', 'col_hispa', 'col_asian', 'col_amind', 'col_pacis', 'col_twora']].sum(axis=1)
filtered_df

state                             college  col_white  col_black  \
0       Alabama            Alabama A & M University   2.762706  95.370244   
1       Alabama            Alabama State University   1.355407  96.806491   
3       Alabama             Athens State University  81.560872  12.450587   
4       Alabama                   Auburn University  86.449566   7.204630   
5       Alabama       Auburn University Main Campus  87.406235   7.883860   
...         ...                                 ...        ...        ...   
2430  Wisconsin       University of Wisconsin-Stout  91.003790   1.327253   
2431  Wisconsin    University of Wisconsin-Superior  89.586359   1.793087   
2432  Wisconsin  University of Wisconsin-Whitewater  85.904589   4.653813   
2433  Wisconsin                  Viterbo University  92.214243   1.859869   
2434  Wisconsin          Wisconsin Lutheran College  86.545339   5.255633   

      col_hispa  col_asian  col_amind  col_pacis  col_twora  dif_white  ...  \
0      0.577441   0.186055   0.157606   0.099440   0.846508 -59.900314  ...   
1      0.907462   0.167595   0.081772   0.054439   0.626834 -54.552670  ...   
3      1.812751   0.762370   2.028190   0.065753   1.319478  15.562983  ...   
4      2.962130   2.238553   0.665542   0.011306   0.468272  25.724310  ...   
5      2.214625   1.847222   0.648058   0.000000   0.000000  25.340940  ...   
...         ...        ...        ...        ...        ...        ...  ...   
2430   2.156830   3.286284   0.378542   0.105612   1.741688  -1.497967  ...   
2431   2.130822   1.235030   2.452742   0.100435   2.701525  -7.024135  ...   
2432   4.547350   1.656402   0.214200   0.057667   2.965980  13.240540  ...   
2433   2.320141   1.794738   0.546329   0.030353   1.234327   1.980485  ...   
2434   4.673462   1.110632   0.596479   0.066274   1.752180  33.975082  ...   

      mkt_amind  mkt_pacis  mkt_twora  public  private  selective  \
0      0.559253   0.052924   1.919729     1.0      0.0        1.0   
1      0.337028   0.037316   1.409747     1.0      0.0        1.0   
3      0.509181   0.046996   1.790873     1.0      0.0        0.0   
4      0.528624   0.043030   1.703866     1.0      0.0        0.0   
5      0.460121   0.098045   1.245337     1.0      0.0        0.0   
...         ...        ...        ...     ...      ...        ...   
2430   1.067838   0.070899   1.509660     1.0      0.0        1.0   
2431   1.862803   0.121511   1.699608     1.0      0.0        1.0   
2432   0.722951   0.044428   2.756883     1.0      0.0        1.0   
2433   0.526834   0.083495   1.601961     0.0      1.0        1.0   
2434   0.361789   0.035315   2.601499     0.0      1.0        1.0   

      more_selective  non_selective  total_enrollment    col_poc  
0                0.0            0.0       4405.666667  97.237294  
1                0.0            0.0       4795.555556  98.644593  
3                0.0            1.0       3082.666667  18.439128  
4                1.0            0.0      21013.250000  13.550434  
5                1.0            0.0      19918.000000  12.593765  
...              ...            ...               ...        ...  
2430             0.0            0.0       8102.666667   8.996210  
2431             0.0            0.0       2405.000000  10.413641  
2432             0.0            0.0      10301.777778  14.095411  
2433             0.0            0.0       2004.555556   7.785757  
2434             0.0            0.0        966.222222  13.454661  

[1596 rows x 30 columns]

Lastly, we must create numerical variables to replace the state and college variables. The Logistic Regression models we run require they are in a numerical format.

Our original code converted the `state` and `college` variables into one-hot vectors; however, the models would not run because a 'Singular matrix' was found, meaning there is a combination of predictor variables that perfectly predicts the outcome variable, resulting in infinite coefficient estimates.

To fix this, decided to label encode them instead. While doing this may imply an ordinal relationship between the `state` or `college` variables and the target variables, since they are listed in alphabetical order and this will be converted to numbers based on this order, we are assuming there is not relationship between the state or college's first letter and the selectivity levels.

In [ ]:
df_sum_hot = pd.get_dummies(filtered_df, columns = ['state'], dtype = int)
df_hot = df_sum_hot.drop(columns = ["college", "total_enrollment"])
# New England
df_hot["new_england"] = df_hot["state_Connecticut"] + df_hot["state_Massachusetts"] + \
df_hot["state_Rhode Island"] + df_hot["state_Maine"] + df_hot["state_New Hampshire"] + df_hot["state_Vermont"]
# Mid East
df_hot["mid_east"] = df_hot["state_New York"] + df_hot["state_New Jersey"] + df_hot["state_Pennsylvania"] + \
df_hot["state_Maryland"] + df_hot["state_Delaware"] + df_hot["state_District of Columbia"]
# South East
df_hot["south_east"] = df_hot["state_Florida"] + df_hot["state_South Carolina"] + \
df_hot["state_Georgia"] + df_hot["state_Tennessee"] + df_hot["state_West Virginia"] + df_hot["state_Virginia"] + \
df_hot["state_North Carolina"] + df_hot["state_Arkansas"] + df_hot["state_Alabama"] + \
df_hot["state_Mississippi"] + df_hot["state_Louisiana"] + df_hot["state_Kentucky"]
#  Great Lakes
df_hot["great_lakes"] = df_hot["state_Ohio"] + df_hot["state_Michigan"] + df_hot["state_Wisconsin"] + \
df_hot["state_Illinois"] + df_hot["state_Indiana"]
# South West
df_hot["south_west"] = df_hot["state_Texas"] + df_hot["state_Arizona"] + df_hot["state_Oklahoma"] + \
df_hot["state_New Mexico"]
# Rocky Mountains does not have Wyoming
df_hot["rocky_mountains"] = df_hot["state_Colorado"] + df_hot["state_Utah"] + df_hot["state_Montana"] + \
df_hot["state_Idaho"]
# Plains
df_hot["plains"] = df_hot["state_Iowa"] + df_hot["state_Missouri"] + df_hot["state_Kansas"] + \
df_hot["state_Nebraska"] + df_hot["state_South Dakota"] + df_hot["state_North Dakota"]  + df_hot["state_Minnesota"]
# Far West
df_hot["far_west"] = df_hot["state_California"] + df_hot["state_Alaska"] + df_hot["state_Hawaii"] + \
df_hot["state_Nevada"] + df_hot["state_Oregon"] + df_hot["state_Washington"]

df_hot = df_hot.drop(columns = ['state_Alabama',
       'state_Alaska', 'state_Arizona', 'state_Arkansas', 'state_California',
       'state_Colorado', 'state_Connecticut', 'state_Delaware',
       'state_District of Columbia', 'state_Florida', 'state_Georgia',
       'state_Hawaii', 'state_Idaho', 'state_Illinois', 'state_Indiana',
       'state_Iowa', 'state_Kansas', 'state_Kentucky', 'state_Louisiana',
       'state_Maine', 'state_Maryland', 'state_Massachusetts',
       'state_Michigan', 'state_Minnesota', 'state_Mississippi',
       'state_Missouri', 'state_Montana', 'state_Nebraska', 'state_Nevada',
       'state_New Hampshire', 'state_New Jersey', 'state_New Mexico',
       'state_New York', 'state_North Carolina', 'state_North Dakota',
       'state_Ohio', 'state_Oklahoma', 'state_Oregon', 'state_Pennsylvania',
       'state_Rhode Island', 'state_South Carolina', 'state_South Dakota',
       'state_Tennessee', 'state_Texas', 'state_Utah', 'state_Vermont',
       'state_Virginia', 'state_Washington', 'state_West Virginia',
       'state_Wisconsin'])

selectivity = ["more_selective", "selective", "non_selective"]
df_hot["selectivity"] = df_hot[selectivity].idxmax(axis=1)
df_hot = df_hot.drop(columns = ["more_selective", "selective", "non_selective"])

df_hot.to_csv("filtered_data.csv")

df_hot

col_white  col_black  col_hispa  col_asian  col_amind  col_pacis  \
0      2.762706  95.370244   0.577441   0.186055   0.157606   0.099440   
1      1.355407  96.806491   0.907462   0.167595   0.081772   0.054439   
3     81.560872  12.450587   1.812751   0.762370   2.028190   0.065753   
4     86.449566   7.204630   2.962130   2.238553   0.665542   0.011306   
5     87.406235   7.883860   2.214625   1.847222   0.648058   0.000000   
...         ...        ...        ...        ...        ...        ...   
2430  91.003790   1.327253   2.156830   3.286284   0.378542   0.105612   
2431  89.586359   1.793087   2.130822   1.235030   2.452742   0.100435   
2432  85.904589   4.653813   4.547350   1.656402   0.214200   0.057667   
2433  92.214243   1.859869   2.320141   1.794738   0.546329   0.030353   
2434  86.545339   5.255633   4.673462   1.110632   0.596479   0.066274   

      col_twora  dif_white  dif_black  dif_hispa  ...    col_poc  new_england  \
0      0.846508 -59.900314  68.393853  -5.453678  ...  97.237294            0   
1      0.626834 -54.552670  58.535882  -3.203888  ...  98.644593            0   
3      1.319478  15.562983 -14.094827  -3.879124  ...  18.439128            0   
4      0.468272  25.724310 -24.169229  -1.976218  ...  13.550434            0   
5      0.000000  25.340940 -24.407380  -1.166189  ...  12.593765            0   
...         ...        ...        ...        ...  ...        ...          ...   
2430   1.741688  -1.497967   0.077551  -0.600468  ...   8.996210            0   
2431   2.701525  -7.024135   0.536015  -0.301343  ...  10.413641            0   
2432   2.965980  13.240540  -5.197005  -4.122425  ...  14.095411            0   
2433   1.234327   1.980485  -0.754611  -1.171221  ...   7.785757            0   
2434   1.752180  33.975082 -10.807611 -14.475443  ...  13.454661            0   

      mid_east  south_east  great_lakes  south_west  rocky_mountains  plains  \
0            0           1            0           0                0       0   
1            0           1            0           0                0       0   
3            0           1            0           0                0       0   
4            0           1            0           0                0       0   
5            0           1            0           0                0       0   
...        ...         ...          ...         ...              ...     ...   
2430         0           0            1           0                0       0   
2431         0           0            1           0                0       0   
2432         0           0            1           0                0       0   
2433         0           0            1           0                0       0   
2434         0           0            1           0                0       0   

      far_west     selectivity  
0            0       selective  
1            0       selective  
3            0   non_selective  
4            0  more_selective  
5            0  more_selective  
...        ...             ...  
2430         0       selective  
2431         0       selective  
2432         0       selective  
2433         0       selective  
2434         0       selective  

[1596 rows x 33 columns]

Now we have our cleaned and transformed dataframe ready for our models in phase3_code! We have exported it as a csv below and called it filtered_data.csv.

In [ ]:
import os

file_path = '/content/drive/My Drive/INFO 4390 Project/filtered_data.csv'

if os.path.exists(file_path):
    # If the file exists, delete it
    os.remove(file_path)
    print("Existing file removed.")

df_hot.to_csv(file_path, index=False, mode='w')

Existing file removed.
